hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2123,6 +2125,8 @@ func GetControlPlaneOperatorImage(ctx context.Context, hc *hyperv1.HostedCluster
 }
 
 func reconcileControlPlaneOperatorDeployment(
+	ctx context.Context,
```

Comment:
Passing a client and context to a reconcile function breaks the general pattern of reconciles where we modify a resource in memory (and also makes it less testable).

My preference is that before calling this reconcile, that you find the ca configmap and if it exists, copy it over to the hcp namespace, and just pass a boolean to this reconcile indicating that you have done so.

Reply:
This should be fixed now.